File: projectFive/java/Machine/src/main/java/besoir/Framework/State.java
package besoir.Framework;

public class State {
    private int parts;
    private double time;
    public State(int parts, double time) {
        this.parts = parts;
        this.time = time;
    }

    public int getParts() {
        return this.parts;
    }

    public double getTime() {
        return this.time;
    }

    public boolean isEmpty() {
        return this.parts == 0;
    }
}

File: projectSix/cpp/State.cpp
#include "state.hpp"

State::State(int q, double s) {
    this->count = q;
    this->time = s;
}

int State::get_count() {
    return this->count;
}

int State::get_time() {
    return this->time;
}

void State::print_state() {
    cout << "Parts: " << to_string(count) << " Time: " << to_string(time) << endl;
}